File: programme/pangram-checker/pangram-checker.js


function isPangram(input){
    const alphabet = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']
    const letters = {}
    let pangram = true

    //add an entry on "letters" object for every character in input
    for(let i=0; i<input.length; i++){
        //convert the read character to lowercase
        char = input.charAt(i).toLowerCase()
        letters[char] = true
    }

    //iterate through the object for every letter in alphabet
    alphabet.forEach( letter => {
        //if the letter is not inside of the "letters" object then set pangram to false
        if(!letters[letter]){
            pangram = false
        }
    })

    return pangram
}

File: programme/pangram-checker/pangram-checker.cpp
// Write a programme to check Pangram sentence
#include <iostream>

using namespace std;

int main()
{
    // Taking an array of character of max size 100
    char str[100];

    // Scanning the input line
    cout << "Input  : ";
    scanf("%[^\n]s", str);

    // isPangram is a variable that store if a sentence is Pangram or not
    char isPangram;

    // As per description, we are checking if a-z is present or not
    for (char i = 'a'; i <= 'z'; i++)
    {
        // Iterating variable for given string
        int j;

        // Iterating through given string
        for (j = 0; str[j] != '\0'; j++)
        {

            // Checking the character in the given sentence in Uppercase (A-Z) or not
            if (str[j] >= 'A' && str[j] <= 'Z')
            {

                // Change the character into Lowercase (a-z) if it is Uppercase (A-Z)
                str[j] += 32;
            }

            // We found ith alphabet
            if (str[j] == i)
            {
                // Breaking from the loop as we
                break;
            }
            // do not have to check more after one alphabet at least once
        }
        // If we reach to the end of the string and still we do not locate an
        if (str[j] == '\0')
        {
            // alphabet in the string, then the sentence will not be a pangram
            isPangram = 'n';

            // As we are sure it is not a pangram sentence, then just getting out from the loop
            break;
        }
    }

    cout << "Output : ";
    if (isPangram == 'n')
    {
        cout << "Not a pangram sentence\n";
    }
    else
    {
        cout << "Pangram Sentence\n";
    }

    return 0;
}
